static/js/jira-connect-ghe.js

```diff
@@ -0,0 +1,62 @@
+const ALLOWED_PROTOCOLS = ["http:", "https:"];
+const ALLOWED_DOMAIN_EXTENSIONS = [".com", ".net"];
+const GITHUB_CLOUD = ["github.com", "www.github.com"];
+
+/**
+ * Method that checks the validity of the passed URL
+ *
+ * @returns {boolean}
+ */
+const checkValidUrl = inputURL => {
+	try {
+		const { protocol, hostname} = new URL(inputURL);
+
+		if (!ALLOWED_PROTOCOLS.some(allowedProtocol => protocol.startsWith(allowedProtocol))) {
+			return false;
+		}
+		if (!ALLOWED_DOMAIN_EXTENSIONS.some(extension => hostname.endsWith(extension))) {
+			return false;
+		}
+		if (GITHUB_CLOUD.some(ghCloud => ghCloud === hostname)) {
+			// TODO: Need to alert the users that this URL is GitHub cloud and not Enterprise
+			// Waiting for design: https://softwareteams.atlassian.net/browse/ARC-1418
+			return false;
+		}
+
+		return true;
+	} catch (e) {
+		return false;
+	}
+};
+
+$("#gheServerURL").on("keyup", event => {
```

Comment:
might need some `debounce` here for better UX, jquery has a plugin, or we could write our own `debounce` (should be easy, just some `setTimeout/clearTimeout`)

--

Or actually it is okay to NOT `debounce`, as this piece of code is small?

Reply:
I think debounce here would be an overkill, as there is no API call nor any major performance affecting things happening. The main function happens once you click the button.